src/Kafka/Producer.purs

```diff
@@ -78,42 +127,113 @@ type MessageImpl =
 -- | https://github.com/tulios/kafkajs/blob/dcee6971c4a739ebb02d9279f68155e3945c50f7/types/index.d.ts#L787
 foreign import data Producer :: Type
 
+-- | * `acks`
+-- |   * Control the number of required acks.
+-- |   * default: `AcksAll`
+-- | * `compression`
+-- |   * compression codec
+-- |   * default: `CompressionTypeNone`
+-- | * `timeout`
+-- |   * The time to await a response in ms
+-- |   * default: `30000`
 -- | * `topicMessages`
 -- |   * a list of topics and for each topic a list of messages
 type ProducerBatch =
-  { topicMessages :: Array TopicMessages
+  { acks :: Data.Maybe.Maybe Acks
+  , compression :: Data.Maybe.Maybe CompressionType
+  , timeout :: Data.Maybe.Maybe Data.Time.Duration.Milliseconds
+  , topicMessages :: Array TopicMessages
   }
 
 -- | https://github.com/tulios/kafkajs/blob/dcee6971c4a739ebb02d9279f68155e3945c50f7/types/index.d.ts#L753
 -- |
 -- | Optional
+-- | * `acks?: number`
+-- | * `compression?: CompressionTypes`
+-- | * `timeout?: number`
 -- | * `topicMessages?: TopicMessages[]`
 type ProducerBatchImpl =
   Kafka.FFI.Object
     ()
-    ( topicMessages :: Array TopicMessagesImpl
+    ( acks :: AcksImpl
+    , compression :: CompressionTypeImpl
+    , timeout :: Number
+    , topicMessages :: Array TopicMessagesImpl
     )
 
+-- | see [Options](https://kafka.js.org/docs/producing#options)
+-- |
+-- | * `allowAutoTopicCreation`
+-- |   * Allow topic creation when querying metadata for non-existent topics
+-- |   * default: `true`
+-- | * `idempotent`
+-- |   * If enabled producer will ensure each message is written exactly once. Acks must be set to `AcksAll`. Retries will default to `MAX_SAFE_INTEGER`.
+-- |   * default: `false`
+-- | * `maxInFlightRequests`
+-- |   * Max number of requests that may be in progress at any time. If `Nothing` then no limit.
+-- |   * default: `Nothing`
+-- | * `metadataMaxAge`
+-- |   * The period of time in milliseconds after which we force a refresh of metadata even if we haven't seen any partition leadership changes to proactively discover any new brokers or partitions
+-- |   * default: `300000`
+-- | * `transactionTimeout`
+-- |   * The maximum amount of time in ms that the transaction coordinator will wait for a transaction status update from the producer before proactively aborting the ongoing transaction. If this value is larger than the `transaction.max.timeout.ms` setting in the broker, the request will fail with a `InvalidTransactionTimeout` error
+-- |   * default: `60000`
```

Comment:
A couple more:

```suggestion
-- |   * default: `Milliseconds 300000.0`
-- | * `transactionTimeout`
-- |   * The maximum amount of time in ms that the transaction coordinator will wait for a transaction status update from the producer before proactively aborting the ongoing transaction. If this value is larger than the `transaction.max.timeout.ms` setting in the broker, the request will fail with a `InvalidTransactionTimeout` error
-- |   * default: `Milliseconds 60000.0`
```

Reply:
I feel like this is not necessary. The type definition (`Data.Time.Duration.Milliseconds`) already has this information and the comments are just supplementary to the type definition.

Reply:
I feel like it's still clearer than not, and doesn't really hurt any. But it's a non-blocking suggestion, this PR is good to go.

Reply:
Updated